src/stream.c

```diff
@@ -33,17 +34,21 @@ struct sender {
 
 /* Receive */
 struct receiver {
-	struct metric *metric; /**< Metrics for receiving            */
-	struct tmr tmr_rtp;   /**< Timer for detecting RTP timeout  */
-	struct jbuf *jbuf;    /**< Jitter Buffer for incoming RTP   */
-	uint64_t ts_last;     /**< Timestamp of last recv RTP pkt   */
-	uint32_t rtp_timeout; /**< RTP Timeout value in [ms]        */
-	uint32_t ssrc;        /**< Incoming synchronization source  */
-	uint32_t pseq;        /**< Sequence number for incoming RTP */
-	bool ssrc_set;        /**< Incoming SSRC is set             */
-	bool pseq_set;        /**< True if sequence number is set   */
-	bool rtp_estab;       /**< True if RTP stream established   */
-	bool enabled;         /**< True if enabled                  */
+	struct metric *metric;         /**< Metrics for receiving            */
+	struct jbuf *jbuf;             /**< Jitter Buffer for incoming RTP   */
+	RE_ATOMIC bool enabled;        /**< True if enabled                  */
+	RE_ATOMIC uint64_t ts_last;    /**< Timestamp of last recv RTP pkt   */
+	RE_ATOMIC uint32_t ssrc;       /**< Incoming synchronization source  */
+	RE_ATOMIC bool ssrc_set;       /**< Incoming SSRC is set             */
+	uint32_t pseq;                 /**< Sequence number for incoming RTP */
+	bool pseq_set;                 /**< True if sequence number is set   */
+	bool rtp_estab;                /**< True if RTP stream established   */
+};
```

Comment:
it looks like around half the struct needs locking. Instead of adding all these atomic fields,
it might be cleaner to just have one mutex that is valid for the whole struct.

A good example here is AUBUF where the API is thread-safe and all the mutex magic
happens inside the .c file

it might be better to create a new file receiver.c and hide the locking inside it,
and add functions for setters and getters.

Reply:
Good idea! This will simplify to understand the thread safety. I'll try this. Although we should be careful to keep the lock time very short.

Reply:
Maybe it would be also good for the understanding if the function that run in the RX thread are also moved to receiver.c. I'll try this.